File: DataStructures/Stacks/atcoder/ A51/A51.py
import sys

input = sys.stdin.readline

Q = int(input())
stack = []
output = []

for _ in range(Q):
    query = input().split()
    if query[0] == '1':
        stack.append(query[1])
    elif query[0] == '2':
        output.append(stack[-1])
    elif query[0] == '3':
        stack.pop()

print('\n'.join(output))

File: DataStructures/Stacks/atcoder/ A51/A51.ts
import * as fs from 'fs';

const input: string[] = fs.readFileSync('/dev/stdin', 'utf8').trim().split('\n');
const Q: number = parseInt(input[0], 10);

const stack: string[] = [];
const output: string[] = [];

for (let i = 1; i <= Q; i++) {
    const line = input[i];
    const [type, x] = line.split(' ');

    if (type === '1') {
        stack.push(x);
    } else if (type === '2') {
        output.push(stack[stack.length - 1]);
    } else if (type === '3') {
        stack.pop();
    }
}

console.log(output.join('\n'));
